scripts/stats.js

```diff
@@ -0,0 +1,143 @@
+// Description
+//   Collects and yields general slack statistics for analysis and insight
+//
+// Commands:
+//   `!stats (rooms|commands)` - Yields general slack statistics for analysis and insight
+
+// Default stats upon reset
+DEFAULT_STATS = {rooms: {}, commands: {}};
+
+/////////////////////
+// HELPER COMMANDS //
+/////////////////////
+
+// Increment counter in map, first setting to 0 if it does not exist
+function incrementCounter(map, entry) {
+    if (!(entry in map)) {
+        map[entry] = 0;
+    }
+    map[entry]++;
+}
+
+// Returns a sorted list of object entries
+function getSortedEntries(object) {
+    var entries = [];
+    for (var entry in object) entries.push([entry, object[entry]])
+    return entries.sort((a, b) => b[1] - a[1]);
+}
+
+// Retrieves stored slack stats, setting them to default values if they do not exist
+function getStats(robot) {
+    var stats = robot.brain.get('stats');
+    if (!stats) {
+        robot.brain.set('stats', DEFAULT_STATS);
+        stats = robot.brain.get('stats');
+    }
+    return stats;
+}
+
+//////////////////////
+// HANDLER COMMANDS //
+//////////////////////
+
+// Handles room stat
+function handleRoomStat(stats, res) {
+    // If room was not a public channel and not a private channel, exit
+    var room = res.message.room;
+    if (room[0] != 'C' && room[0] != 'G') return;
+    incrementCounter(stats.rooms, room);
+}
+
+// Handles command stat 
+function handleCommandStat(stats, res) {
+    // If the command is not actually a command, exit
+    var command = res.message.text;
+    if (command[0] != '!') {
+        return;
+    }
+
+    incrementCounter(stats.commands, command);
+
+    // If command did not contain any options, exit
+    var baseCommand = command.split(' ')[0];
+    if (baseCommand == command) {
```

Comment:
could you do `if(command.indexOf(' ') < 0)` instead? (and same with other places which you check in a same way.

Reply:
In the instance you mentioned I actually use the split to get baseCommand, so its necessary-- but I'll apply that change to the other location :)